tedge/src/cli/disconnect/bridge.rs

```diff
@@ -0,0 +1,107 @@
+use crate::cli::connect::TEDGE_BRIDGE_CONF_DIR_PATH;
+use crate::command::{Command, ExecutionContext};
+use crate::services::{
+    mosquitto::MosquittoService, tedge_mapper::TedgeMapperService, SystemdService,
+};
+use crate::utils::paths;
+use crate::utils::users::{UserManager, ROOT_USER};
+use which::which;
+
+use super::error::DisconnectBridgeError;
+
+#[derive(Debug)]
+pub struct DisconnectBridge {
+    pub(crate) config_file: String,
+    cloud_name: String,
+    use_mapper: bool,
+}
+
+impl Command for DisconnectBridge {
+    fn description(&self) -> String {
+        format!("remove the bridge to disconnect {} cloud", self.cloud_name)
+    }
+
+    fn execute(&self, context: &ExecutionContext) -> Result<(), anyhow::Error> {
+        match self.stop_bridge(&context.user_manager) {
+            Ok(()) | Err(DisconnectBridgeError::BridgeFileDoesNotExist) => Ok(()),
+            Err(err) => Err(err.into()),
+        }
+    }
+}
+
+impl DisconnectBridge {
+    fn stop_bridge(&self, user_manager: &UserManager) -> Result<(), DisconnectBridgeError> {
+        // If this fails, do not continue with applying changes and stopping/disabling tedge-mapper.
+        self.remove_bridge_config_file()?;
+
+        // Ignore failure
+        let _ = self.apply_changes_to_mosquitto(user_manager);
+
+        // Only C8Y changes the status of tedge-mapper
+        if self.use_mapper && which("tedge_mapper").is_ok() {
+            self.stop_and_disable_tedge_mapper(user_manager);
+        }
+
+        Ok(())
+    }
+
+    fn remove_bridge_config_file(&self) -> Result<(), DisconnectBridgeError> {
+        // Check if bridge exists and stop with code 0 if it doesn't.
+        let bridge_conf_path =
+            paths::build_path_for_sudo_or_user(&[TEDGE_BRIDGE_CONF_DIR_PATH, &self.config_file])?;
```

Comment:
So I would inject `bridge_conf_path` into the `DisconnectBridge` and remove `config_file`.